### PR TITLE
aronmassey/ztest api comment docs

### PR DESCRIPTION
**Pull request by aaronemassey**
Add/update docstrings for ztest on zassume and ztest suite setup

### CODE REVIEW

**Review by asemjonovs — DISMISSED** [review]

**Comment by aaronemassey** [comment]
DNM Due to conversation going on at https://github.com/zephyrproject-rtos/zephyr/pull/48747

File: subsys/testsuite/ztest/include/zephyr/ztest_test_new.h [hunk 1]
```diff
@@ -53,6 +53,10 @@ struct ztest_suite_stats {
 /**
  * Setup function to run before running this suite
  *
+ * Note: Never zassert in the setup. As this will kill the main test thread causing the test process
+ * to hang. Instead, use zassume (for behavior tested elsewhere, see the zassume function doc) or
+ * add the zassert as its own test or as part of the before/after functions.
```
Comment (asemjonovs) [inline subsys/testsuite/ztest/include/zephyr/ztest_test_new.h:58]: This is behavior is being changed with https://github.com/zephyrproject-rtos/zephyr/pull/48768 to allow asserting anywhere. Please update comment to reflect this.

**Review by asemjonovs — CHANGES_REQUESTED** [review]

File: subsys/testsuite/ztest/include/zephyr/ztest_test_new.h [hunk 2]
```diff
@@ -53,6 +53,10 @@ struct ztest_suite_stats {
 /**
  * Setup function to run before running this suite
  *
+ * Note: Never zassert in the setup. As this will kill the main test thread causing the test process
```
Comment (alevkoy) [inline subsys/testsuite/ztest/include/zephyr/ztest_test_new.h:56]: "setup, as"
"thread, causing"

File: subsys/testsuite/ztest/include/zephyr/ztest_test_new.h [hunk 3]
```diff
@@ -53,6 +53,10 @@ struct ztest_suite_stats {
...
+ * to hang. Instead, use zassume (for behavior tested elsewhere, see the zassume function doc) or
```
Comment (alevkoy) [inline subsys/testsuite/ztest/include/zephyr/ztest_test_new.h:57]: This should probably be some kind of Doxygen link.

**Review by alevkoy — DISMISSED** [review]

File: subsys/testsuite/ztest/include/zephyr/ztest_test_new.h (hunk 1 above)
Comment (aaronemassey) [reply subsys/testsuite/ztest/include/zephyr/ztest_test_new.h]: Updated

File: subsys/testsuite/ztest/include/zephyr/ztest_assert.h [hunk 4]
```diff
@@ -141,6 +141,25 @@ static inline bool z_zassume(bool cond, const char *default_msg, const char *fil
...
+ * @brief Skip the test, if @a cond is false
+ *
+ * You probably don't need to call this macro directly. You should
+ * instead use zassume_{condition} macros below.
+ *
+ * Note that when CONFIG_MULTITHREADING=n macro returns from the function. It is
+ * then expected that in that case ztest asserts will be used only in the
+ * context of the test function.
+ *
+ * NOTE: zassume should not be used in replace-of zassert, the goal of zassume is to skip tests that
```
[the remark this answers is not in the document]
Comment (aaronemassey) [reply subsys/testsuite/ztest/include/zephyr/ztest_assert.h]: Done

File: subsys/testsuite/ztest/include/zephyr/ztest_assert.h [hunk 5]
```diff
@@ -141,6 +141,25 @@ static inline bool z_zassume(bool cond, const char *default_msg, const char *fil
 		}                                                                                  \
 	} while (0)
 
+/**
+ * @brief Skip the test, if @a cond is false
+ *
+ * You probably don't need to call this macro directly. You should
+ * instead use zassume_{condition} macros below.
+ *
+ * Note that when CONFIG_MULTITHREADING=n macro returns from the function. It is
```
[the remark this answers is not in the document]
Comment (aaronemassey) [reply subsys/testsuite/ztest/include/zephyr/ztest_assert.h]: Combined It is as It's

File: subsys/testsuite/ztest/include/zephyr/ztest_test_new.h (hunk 2 above)
Comment (aaronemassey) [reply subsys/testsuite/ztest/include/zephyr/ztest_test_new.h]: Removed doc as it's no longer pertinent

File: subsys/testsuite/ztest/include/zephyr/ztest_test_new.h (hunk 3 above)
Comment (aaronemassey) [reply subsys/testsuite/ztest/include/zephyr/ztest_test_new.h]: Removed doc as it's no longer pertinent

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@aaronemassey Is this still supposed to be marked DNM?

**Comment by aaronemassey** [comment]
> @aaronemassey Is this still supposed to be marked DNM?

Whoops, no I forgot to remove the DNM label. Thanks, Henrik!

**Review by asemjonovs — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**